### ISSUE TITLE
[package] GDAL/All: Addition of option: RENAME_INTERNAL_SHAPELIB_SYMBOLS

### ISSUE THREAD

**Issue by JustCallMeRay** [conan-io/conan-center-index#25175]
### Description

Not really a bug (a feature request for a package):
For my specific use case I need to have shapelib exported from GDAL with non-renamed symbols (the default is to rename them to avoid issues with multiply defined symbols). 

Building from source this can be done by adding the `RENAME_INTERNAL_SHAPELIB_SYMBOLS` to the cmake invocation (with `-D`) so addition should be very easy. Configuration is invalid if not using shapelib or using external shapelib 

This functionality still sees active development on the GDAL side as shown by [this](https://github.com/OSGeo/gdal/issues/8532) bug report: 

### Package and Environment Details

* Package Name/Version: **GDAL/all**
* Operating System+version: **N/A**
* Compiler+version: **N/A**
* Docker image: **N/A**
* Conan version: **N/A**
* Python version: **N/A**

### Conan profile

Profile not relevant

### Steps to reproduce

Specify option: `GDAL:RENAME_INTERNAL_SHAPELIB_SYMBOLS` set to False in your preferred way then install the GDAL package with `--build=missing` (as it is unlikely to be built).

### Logs

<details><summary>Click to expand log</summary>

```

```

</details>

**Comment by ErniGH** [comment]
Hey @JustCallMeRay,

I apologize for the delay in getting back to you on this issue. I also saw your other open PR. GDAL already has a lot of options, and adding more and more can be risky since our CI can’t test all possible combinations.

For cases like this, where the option just adds a variable, using `tools.cmake.cmaketoolchain:extra_variables` is usually the easiest way to go. For example, if you want to set `RENAME_INTERNAL_SHAPELIB_SYMBOLS` to `False`, it would look like this:
`-c tools.cmake.cmaketoolchain:extra_variables='{"RENAME_INTERNAL_SHAPELIB_SYMBOLS": "False"}'`

If you want to learn more about global variables, you can check it out here:
https://docs.conan.io/2/reference/config_files/global_conf.html

I tested it locally with version 3.8.3, and it worked fine for me. Let me know if it works for you!

**Comment by JustCallMeRay** [comment]
Hiya,

Sorry for such a late reply.

I did not know you could pass arguments directly to the toolchain which completely solves my problem.

I have tested it locally and everything worked as expected. I had not intended for your CI to run it frequently as I am aware that this is not a commonly used feature of GDAL.

TLDR: I am happy for this to be either merged or closed,

Thanks for getting back to me!
